From 7f2868ceb718ed61b69ebcbd0b6dfae484cabfde Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Morten=20S=C3=B8rvig?= <morten.sorvig@qt.io>
Date: Fri, 19 Sep 2025 12:55:41 +0200
Subject: [PATCH 190/553] wasm: remove wasm-jspi memory limit workaround

Emscripten bug #23834 has been fixed, so the 2GB memory
limit workaround for JSPI is no longer needed.

Change-Id: I8d8f12dba428ccb0e9b75164c7fd9ef1a1332c26
Reviewed-by: Even Oscar Andersen <even.oscar.andersen@qt.io>
Reviewed-by: Lorn Potter <lorn.potter@qt.io>
(cherry picked from commit 8cf3690ed06bc3c48101832420f50ba960348658)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 mkspecs/features/wasm/wasm.prf  | 6 +-----
 src/corelib/Qt6WasmMacros.cmake | 7 +------
 2 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/mkspecs/features/wasm/wasm.prf b/mkspecs/features/wasm/wasm.prf
index a257f5f5cb5..4bbba8c2aa6 100644
--- a/mkspecs/features/wasm/wasm.prf
+++ b/mkspecs/features/wasm/wasm.prf
@@ -50,11 +50,7 @@ exists($$QMAKE_QT_CONFIG) {
     }
     EMCC_LFLAGS += -s INITIAL_MEMORY=$$INITIAL_MEMORY
     isEmpty(QT_WASM_MAXIMUM_MEMORY) {
-        qtConfig(wasm-jspi) {
-            MAXIMUM_MEMORY = 2GB # Avoid triggering Emscripten bug #23834
-        } else {
-            MAXIMUM_MEMORY = 4GB # 32-bit max
-        }
+        MAXIMUM_MEMORY = 4GB # 32-bit max
     } else {
         MAXIMUM_MEMORY = $$QT_WASM_MAXIMUM_MEMORY
     }
diff --git a/src/corelib/Qt6WasmMacros.cmake b/src/corelib/Qt6WasmMacros.cmake
index e185ef67490..032a28f0473 100644
--- a/src/corelib/Qt6WasmMacros.cmake
+++ b/src/corelib/Qt6WasmMacros.cmake
@@ -130,12 +130,7 @@ function(_qt_internal_wasm_add_target_helpers target)
         if(_tmp_maximumMemory)
             set(QT_WASM_MAXIMUM_MEMORY "${_tmp_maximumMemory}")
         elseif(NOT DEFINED QT_WASM_MAXIMUM_MEMORY)
-            if(QT_FEATURE_wasm_jspi)
-                # Work around Emscripten >2GB and JSPI compatibility issue.
-                set(QT_WASM_MAXIMUM_MEMORY "2GB")
-            else()
-                set(QT_WASM_MAXIMUM_MEMORY "4GB")
-            endif()
+            set(QT_WASM_MAXIMUM_MEMORY "4GB")
         endif()
         target_link_options("${target}" PRIVATE "SHELL:-s MAXIMUM_MEMORY=${QT_WASM_MAXIMUM_MEMORY}")
 
-- 
2.51.2

